File: SDKs/adyen.com/BalancePlatformService/1/typescript/src/sdk/models/operations/getlegalentitiesid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetLegalEntitiesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class GetLegalEntitiesIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=http;subtype=basic" })
  basicAuth?: shared.SchemeBasicAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetLegalEntitiesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetLegalEntitiesIdPathParams;

  @SpeakeasyMetadata()
  security: GetLegalEntitiesIdSecurity;
}


export class GetLegalEntitiesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  legalEntity?: any;

  @SpeakeasyMetadata()
  serviceError?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/adyen.com/BalancePlatformService/1/python/src/sdk/models/operations/get_legalentities_id.py
import dataclasses
from typing import Any,Optional
from ..shared import security as shared_security
from ..shared import security as shared_security


@dataclasses.dataclass
class GetLegalEntitiesIDPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetLegalEntitiesIDSecurity:
    api_key_auth: Optional[shared_security.SchemeAPIKeyAuth] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    basic_auth: Optional[shared_security.SchemeBasicAuth] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclasses.dataclass
class GetLegalEntitiesIDRequest:
    path_params: GetLegalEntitiesIDPathParams = dataclasses.field()
    security: GetLegalEntitiesIDSecurity = dataclasses.field()
    

@dataclasses.dataclass
class GetLegalEntitiesIDResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    legal_entity: Optional[Any] = dataclasses.field(default=None)
    service_error: Optional[Any] = dataclasses.field(default=None)
    
